src/monocle_apptrace/wrap_common.py

```diff
@@ -90,7 +101,15 @@ def task_wrapper(tracer: Tracer, to_wrap, wrapped, instance, args, kwargs):
     elif to_wrap.get("span_name"):
         name = to_wrap.get("span_name")
     else:
-        name = f"langchain.task.{instance.__class__.__name__}"
+        name = get_fully_qualified_class_name(instance)
+
+    if 'haystack.core.pipeline.pipeline' in to_wrap['package']:
+        embedding_model = get_embedding_model_haystack(instance)
+        set_embedding_model(embedding_model)
+        inputs = set()
```

Comment:
What does this do?

Reply:
we are getting embedding model in pipeline package , but we need embedding in retriever package , thats why we are getting and setting the embedding model in pipeline package to utilize in retriever package